terraform/gcp/security/gcp-compute-project-os-login.yaml

```diff
@@ -0,0 +1,28 @@
+rules:
+- id: gcp-compute-project-os-login
+  patterns:
+    - pattern: resource
+    - pattern-inside: |
+        resource "google_compute_project_metadata" "..." {
+        ...
+        }
+    - pattern-not-inside: |
+        resource "google_compute_project_metadata" "..." {
+        ...
+        metadata = {
+            enable-oslogin = "True"

```

Comment:
Any idea why the value for this resource is a string, and the value for the instance version of this rule is a boolean? 🤔

Reply:
According to [setCommonInstanceMetadata ](https://cloud.google.com/compute/docs/reference/rest/v1/projects/setCommonInstanceMetadata) JSON representation, fields expect String. Examples I've encountered regarding `enable-oslogin` were also using String ([example1](https://yuweisung.medium.com/applying-gcp-os-login-to-terraform-and-ansible-da1d78c016b5), [example2](https://binx.io/nl/2022/01/28/how-to-use-os-login-for-ssh-access-to-vms-on-gcp/)). 

However, [this](https://aquasecurity.github.io/tfsec/v1.6.2/checks/google/compute/project-level-oslogin/) also shows that it can use boolean. Do you think we should cover the boolean case just to be sure? If yes, how can we change the condition to capture both?

Reply:
I guess it won't hurt to cover both cases 🤷 

The easiest way is to add another pattern-not-inside clause for the boolean case. Negative clauses just act like filters, so they will work even if they're on the same level.

Reply:
Added the use case. Pipeline fails on irrelevant rule (/src/javascript/angular/security/detect-angular-element-methods.js)